src/sentry/tasks/store.py

```diff
@@ -63,6 +65,12 @@ def _do_preprocess_event(cache_key, data, start_time, event_id, process_event):
     })
 
     if should_process(data):
+        # save another version of data for some projects to generate
+        # preprocessing hash that won't be modified by pipeline
+        if data.get('event_id') and int(project) % 5 == 0:
```

Comment:
- Are there cases where `event_id` is actually null?
- Coercing `project` to an integer tripped me out for a few minutes (I thought it was a `project` instance at first, and this would error) — it might be worth going back and changing the local name to `project_id` instead.
- Curious to get @mattrobenolt's take on whether we need to throttle this out of the gate (with the project hash, or just a `random.random()` threshold or something) or if its ok to deploy with it on full blast.

Reply:
so this test fails if i don't check for `event_id`: https://github.com/getsentry/sentry/blob/master/tests/sentry/tasks/test_store.py#L40

I can change that test if there should always be an event_id, just wasn't totally sure if there was some case in which it wouldn't exist.

and yeah, agree about the project thing

Reply:
This made me assume that the `event_id` is required, but maybe that is an incorrect assumption: https://github.com/getsentry/sentry/blob/bef4f3e9cc0becd6218289ee74b168fabb908e1a/src/sentry/coreapi.py#L822-L823

Reply:
> Curious to get @mattrobenolt's take on whether we need to throttle this out of the gate (with the project hash, or just a random.random() threshold or something) or if its ok to deploy with it on full blast.

My body is ready. Full blast it. 💯

Reply:
😎